src/android/Geolocation.java

```diff
@@ -206,7 +310,32 @@ private void clearWatch(JSONArray args, CallbackContext callbackContext) {
             }
             else {
                 this.locationContexts.delete(requestId);
-                fusedLocationClient.removeLocationUpdates(lc.getLocationCallback());
+                fusedLocationClientGms.removeLocationUpdates(lc.getLocationCallback());
+
+                PluginResult result = new PluginResult(PluginResult.Status.OK);
+                callbackContext.sendPluginResult(result);
+            }
+        }
+    }
+
+    /**
+     * Clear Watch Locations in devices which doesn't contains google services (Huawei Devices)
+     * @param callbackContext the callbackContext
+     */
+    private void clearWatchHMS(JSONArray args, CallbackContext callbackContext) {
+        String id = args.optString(0);
+
+        if(id != null) {
+            int requestId = id.hashCode();
+            LocationContext lc = locationContexts.get(requestId);
+
+            if(lc == null) {
+                PluginResult result = new PluginResult(PluginResult.Status.ERROR, LocationError.WATCH_ID_NOT_FOUND.toJSON());
+                callbackContext.sendPluginResult(result);
```

Comment:
This line exists for both cases of the if statement. Maybe it could be passed outside in order to avoid replication? The same could be applied to the `result` variable declaration.